### PR TITLE
refactor(devtools): improving type safety

### PR DESCRIPTION
**Pull request by ducin**
This PR reduces the number of unnecessary `any` occurrences in devtools packages.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit

## PR Type
What kind of change does this PR introduce?

- [X] Refactoring (no functional changes, no api changes)

## What is the current behavior?

There are lots of `any` and `as any` statements in devtools packages.

## What is the new behavior?

As a part of a longer journey together with @JeanMeche, step by step, this PR improves type safety.

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

File: devtools/projects/shell-browser/src/app/same-page-message-bus.ts [hunk 1]
```diff
@@ -10,15 +10,17 @@ import {Events, MessageBus, Parameters} from 'protocol';
 
 type AnyEventCallback<Ev> = <E extends keyof Ev>(topic: E, args: Parameters<Ev[E]>) => void;
 
-export class SamePageMessageBus extends MessageBus<Events> {
-  private _listeners: any[] = [];
+type ListenerFn = (e: MessageEvent) => void
+
+    export class SamePageMessageBus extends MessageBus<Events> {
```
Comment (dgp1130) [inline devtools/projects/shell-browser/src/app/same-page-message-bus.ts:15]: Format: Seems like this got unnecessarily indented.

**Review by dgp1130 — APPROVED** [review]
Thanks for this! I'm all in favor of `unknown` over `any`.

@AleksanderBodurri, all these types look good to you?

File: devtools/projects/shell-browser/src/app/same-page-message-bus.ts (hunk 1 above)
Comment (ducin) [reply devtools/projects/shell-browser/src/app/same-page-message-bus.ts]: @dgp1130 absolutely! thanks for catching. Fixed.

**Comment by ducin** [comment]
@dgp1130 thank you, will be happy to provide all of those type safety improvements! 🔥

**Review by JeanMeche — APPROVED** [review]
LGTM, thx for helping us improve the devtools codebase @ducin !!

**Comment by ducin** [comment]
@AleksanderBodurri would you mind taking a look?

**Review by AleksanderBodurri — APPROVED** [review]
Good stuff, thank you 🙏  Can we do a rebase to resolve the conflicts?

**Comment by JeanMeche** [comment]
Just pushed the rebase

**Review by AleksanderBodurri — APPROVED** [review]

**Review by AleksanderBodurri — APPROVED** [review]

**Comment by AleksanderBodurri** [comment]
Looks like theres some CI failures @ducin

**Review by AleksanderBodurri — APPROVED** [review]

**Comment by dylhunn** [comment]
@ducin Looks like this actually needs a rebase.

**Comment by ducin** [comment]
@dylhunn done, @pkozlowski-opensource I hope we can remove `action: cleanup` label

**Comment by pkozlowski-opensource** [comment]
This PR was merged into the repository by commit 4be253483d045cfee6b42766c9dfd8c9888057e0.

**Pull request closed without merging**